infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -203,109 +172,117 @@ private CorfuRuntime getNewCorfuRuntime() {
             return false;
         }
+
         return true;
     }
 
     /**
      * Forward an orchestrator request to the orchestrator service.
      *
-     * @param msg corfu message containing ORCHESTRATOR_REQUEST
-     * @param ctx netty ChannelHandlerContext
-     * @param r   server router
+     * @param req  a message containing an ORCHESTRATOR request
+     * @param ctx  the netty ChannelHandlerContext
+     * @param r    the server router
      */
-    @ServerHandler(type = CorfuMsgType.ORCHESTRATOR_REQUEST)
-    public synchronized void handleOrchestratorMsg(@Nonnull CorfuPayloadMsg<OrchestratorMsg> msg,
-                                                   @Nonnull ChannelHandlerContext ctx,
-                                                   @Nonnull IServerRouter r) {
-        log.debug("Received an orchestrator message {}", msg);
-        orchestrator.handle(msg, ctx, r);
+    @RequestHandler(type = PayloadCase.ORCHESTRATOR_REQUEST)
+    public synchronized void handleOrchestratorMsg(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
+        log.debug("handleOrchestratorMsg: message:{}", req.getPayload().getOrchestratorRequest());
+        orchestrator.handle(req, ctx, r);
     }
 
     /**
      * Bootstraps the management server.
-     * The msg contains the layout to be bootstrapped.
+     * The msg contains the Layout to be bootstrapped with.
      *
-     * @param msg corfu message containing MANAGEMENT_BOOTSTRAP_REQUEST
-     * @param ctx netty ChannelHandlerContext
-     * @param r   server router
+     * @param req  a message containing a BOOTSTRAP_MANAGEMENT request
+     * @param ctx  the netty ChannelHandlerContext
+     * @param r    the server router
      */
-    @ServerHandler(type = CorfuMsgType.MANAGEMENT_BOOTSTRAP_REQUEST)
-    public synchronized void handleManagementBootstrap(CorfuPayloadMsg<Layout> msg,
-                                                       ChannelHandlerContext ctx, IServerRouter r) {
-        if (serverContext.getManagementLayout() != null) {
-            // We are already bootstrapped, bootstrap again is not allowed.
-            log.warn("handleManagementBootstrap: Got a request to bootstrap a server " +
-                    "with {} which is already bootstrapped, rejecting!", msg.getPayload());
-            r.sendResponse(ctx, msg, new CorfuMsg(CorfuMsgType.MANAGEMENT_ALREADY_BOOTSTRAP_ERROR));
+    @RequestHandler(type = PayloadCase.BOOTSTRAP_MANAGEMENT_REQUEST)
+    public synchronized void handleBootstrapManagement(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
+        final Layout layout = getLayout(req.getPayload().getBootstrapManagementRequest().getLayout());
+        HeaderMsg responseHeader;
+        ResponseMsg response;
+
+        if (layout == null || layout.getClusterId() == null) {
+            log.warn("handleBootstrapManagement[{}]: Got a request to bootstrap a server with " +
```

Comment:
Can we improve the logging message here, maybe just "handleBootstrapManagement[{}]: incomplete layout {}" would suffice.

Reply:
Also, the log level should be `log.error`

Reply:
Done.